src/Api/CollectorApiClient.php

```diff
@@ -0,0 +1,156 @@
+                    'Full-Sync-Requested' => $fullSyncRequested ? '1' : '0',
+                    'User-Agent' => 'ps-eventbus/' . $this->module->version,
+                ],
+            ],
+            $multipartBody->getContents()
+        );
+
+        // Send request and parse response
+        $rawResponse = $this->client->sendRequest($request);
+        $jsonResponse = json_decode($rawResponse->getBody()->getContents(), true);
+        $response = [
+            'status' => substr((string) $rawResponse->getStatusCode(), 0, 1) === '2',
+            'httpCode' => $rawResponse->getStatusCode(),
+            'body' => $jsonResponse,
+            'upload_url' => $request->getUri(),
+        ];
+
+        return $response;
+    }
+
+    /**
+     * Push information about removed ShopContents to CloudSync
+     *
+     * @param string $jobId
+     * @param string $data
+     * @param int $startTime in seconds since epoch
+     *
+     * @return array
+     */
+    public function uploadDelete(string $jobId, string $data, int $startTime)
+    {
+        // Prepare request
+        $file = new PostFileApi('file', $data, 'file');
+        $multipartBody = new MultipartBody([], [$file], Config::COLLECTOR_MULTIPART_BOUNDARY);
+        $request = new Request(
+            'POST',
+            '/delete/' . $jobId,
+            [
+                'timeout' => $this->getRemainingTime($startTime),
+                'headers' => [
+                    'Accept' => 'application/json',
+                    'Authorization' => "Bearer $this->jwt",
+                    'Content-Length' => $file->getContent()->getSize(),
+                    'Content-Type' => 'multipart/form-data; boundary=' . Config::COLLECTOR_MULTIPART_BOUNDARY,
+                    'User-Agent' => 'ps-eventbus/' . $this->module->version,
+                ],
+            ],
+            $multipartBody->getContents()
+        );
+
+        // Send request and parse response
+        $rawResponse = $this->client->sendRequest($request);
+        $jsonResponse = json_decode($rawResponse->getBody()->getContents(), true);
+        $response = [
+            'status' => substr((string) $rawResponse->getStatusCode(), 0, 1) === '2',
+            'httpCode' => $rawResponse->getStatusCode(),
+            'body' => $jsonResponse,
+            'upload_url' => $request->getUri(),
+        ];
+
+        return $response;
+    }
+
+    /**
+     * Get the remaining time of execution for the request. We keep a margin
+     * of 1.5s to parse and answser our own client
+     *
+     * @param int $startTime in seconds since epoch
+     *
+     * @return float
+     */
+    public function getRemainingTime(int $startTime)
+    {
+        $remainingTime = time() - $startTime;
+        if ($remainingTime <= 0) {
+            return 0;
+        }
+
+        return $remainingTime - Config::COLLECTOR_MAX_TIMEOUT - 1.5;
```

Comment:
Pourquoi 1.5 🤔

Reply:
In the worst case the timeout will be fully consumed, and no more time will be left for the JSON processing and state storage in the MySQL database.

Reply:
Just added more comments on this part, let me know if this is anyway better like this :)